### DISCUSSION TITLE
More I/Os on Seeed's XIAO

### DISCUSSION THREAD

**Discussion by limengdu** [Seeed-Studio/OSHW-XIAO-Series#2] [In Development]
XIAO is Seeed Studio’s most popular thumb-sized, powerful microcontroller units (MCUs), tailor-made for space-conscious projects that require high performance and wireless connectivity. Over the years, the XIAO family has expanded to include eight core boards, featuring mainstream hardware platforms such as ESP32, RP2040, nRF52840, and SAMD21. As user applications become increasingly diverse, users need more than just XIAO’s mini size; they also require more chip I/O interface expansion capabilities. Currently, the 14 pins seem somewhat insufficient. Therefore, we are now looking for a clever way to expand the number of I/Os to meet the more complex and varied user product design needs.

Today, we're proposing an expansion method and hope that the enthusiastic community will provide valuable feedback.

![image](https://github.com/Seeed-Projects/SeeedStudio_XIAO_Series/assets/37475446/2fd564ee-7fbf-4d8e-a1db-7dc54d28491d)

- Expand from the bottom of the XIAO mainboard.

- An additional pads will be added to the bottom. These pads can be used for SMT (Surface Mount Technology) soldering, or can be soldered using a soldering iron.

- When users need to utilize these pads, they can design their base PCB with a hole such that half of the module’s bottom pads are exposed. This allows users to manually solder using a soldering iron. If the user's base PCB is designed for SMT soldering of components, then half of these bottom pads can also be soldered to the baseboard.

- When users only need the 14 pins on board edge for their requirements, the extra pads on the bottom do not interfere. The user's baseboard can also be designed without holes.

- The spacing between the bottom pads can be 2.54mm, which is more solder-friendly, although this leads to fewer pads being led out. If the spacing is reduced to 2.0mm, more pads can be accommodated.

**Comment by pistachio101** [comment]
2.00mm looks more convincing, what will be those pin definitions?

**Reply by limengdu** [reply]
Maybe more ADCs or UARTs, it may depend on what interfaces we have left on the chip that we are not using.

**Comment by kevin-seeed** [comment]
Quick update on the demo board, show how it works. easy to solder.
![image](https://github.com/Seeed-Studio/XIAO_Series/assets/170942971/83efb78d-ba81-4b4b-9011-7f9c5e1cbd5d)

https://github.com/Seeed-Studio/XIAO_Series/assets/170942971/f5a70026-a65e-4459-8312-e0309199b1ee

**Comment by PjGLasso** [comment]
![20240529_144528](https://github.com/Seeed-Studio/XIAO_Series/assets/144751661/7061cbaf-47ff-4407-8f1b-c6d0c2606c0e)
![20240529_144431](https://github.com/Seeed-Studio/XIAO_Series/assets/144751661/dce3b7d0-8d1d-48d6-8146-f4efc4c80f6f)
![20240529_144516](https://github.com/Seeed-Studio/XIAO_Series/assets/144751661/21d042ca-6d6d-447f-a85f-65063c27baff)

**12**

https://forum.seeedstudio.com/t/xiao-io-expander-coming/272541/27?u=pj_glasso

**Comment by StuartsProjects** [comment]
The extra pins underneath is definetly a good idea. 

If you want to use an RF device such as LoRa, then that can need 7 IO pins, SCK, MOSI, MISO, NSS, NRESET, BUSY, DIO1. 

That does not leave many pins for other stuff, such as displays or sensors etc. 

2mm spacing allows for more pins and is not that much more difficult to solder than 2.54mm. 

However, looking at the bottom of a XIAO, would this arrangement of pads underneath be possible;

![Screenshot - 02_06_2024 , 16_54_38](https://github.com/Seeed-Studio/XIAO_Series/assets/28686802/98a9b509-baa2-4c6b-afc6-474530799742)

Thats 16 pins, and it would be great if the battery connections was one of them, avoids soldering a wire to a pad. 

You would then have the possibility of 27 IO pins !

Stuart Robinson

[https://stuartsprojects.github.io/](url)

**Comment by Peterhj05** [comment]
Hi. A brilliant Idea! What is the estimated release time for these new XIAO's and what vesrion will be done? Appreciated. Peter

**Reply by limengdu** [reply]
If all goes well, it'll be in July.

**Reply by StuartsProjects** [reply]
> If all goes well, it'll be in July.

Its September ..................

**Comment by kevin-seeed** [comment]
Two new XIAOs have 8 more pins
https://www.seeedstudio.com/Seeed-XIAO-RP2350-p-5944.html
https://www.seeedstudio.com/Seeed-XIAO-RA4M1-p-5943.html
![image](https://github.com/user-attachments/assets/a400aeae-9f7a-486b-99ca-29500197a619)
![image](https://github.com/user-attachments/assets/68b6cfa0-adf0-4abe-bfa0-c2dfdd903843)

**Reply by StuartsProjects** [reply]
Is that likely to be the layout if the extra IO idea is spread to boards such as the XIAO ESP32S3 or nrf52840 modules ?

I did do a PCB, for the ESP32S3 Sense, that had large pads with 1.4mm holes in the locations for the BAT+ and EN pads and I was then able to surface mount the ESP32S3 and solder connect to those pads. 

Since I dont need access to the MTCK, MTDO, MTDI and MTMS pins for programming once the board is soldered in place on the PCB would there be any issues with adding pads for those 4 pins so they can be used for general purpose IO ?

**Reply by PjGLasso** [reply]
Sure can. I have demo on the forum using the pins for IO. (LED)
